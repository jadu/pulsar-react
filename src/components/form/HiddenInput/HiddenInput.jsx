import React from 'react';

export default class HiddenInput extends React.Component {

  static defaultProps = {
    className: 'form__control',
    type: 'hidden'
  };

  render() {
    let {
      ariaDescribedby,
      id,
      idGuid,
      ...props
    } = this.props;

    return (
      <input 
        id={id ? id : idGuid} 
        aria-describedby={ariaDescribedby}
        {...props} 
      />
    );
  }
}