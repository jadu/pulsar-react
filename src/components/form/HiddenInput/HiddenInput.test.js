'use strict';

import React from 'react';
import { render } from 'enzyme';

import HiddenInput from '../HiddenInput/HiddenInput';

it('renders the basic input', () => {
  const tree = render(
    <HiddenInput />
  );
  expect(tree).toMatchSnapshot();
});

it('renders the basic input with id', () => {
  const tree = render(
    <HiddenInput id="foo" />
  );
  expect(tree).toMatchSnapshot();
});

it('renders the basic input with idGuid', () => {
  const tree = render(
    <HiddenInput idGuid="bar" />
  );
  expect(tree).toMatchSnapshot();
});

it('will prefer id over idGuid', () => {
  const tree = render(
    <HiddenInput id="foo" idGuid="bar" />
  );
  expect(tree).toMatchSnapshot();
});
