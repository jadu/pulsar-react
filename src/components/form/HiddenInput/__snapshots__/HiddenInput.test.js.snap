// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the basic input 1`] = `
<input
  class="form__control"
  type="hidden"
/>
`;

exports[`renders the basic input with id 1`] = `
<input
  class="form__control"
  id="foo"
  type="hidden"
/>
`;

exports[`renders the basic input with idGuid 1`] = `
<input
  class="form__control"
  id="bar"
  type="hidden"
/>
`;

exports[`will prefer id over idGuid 1`] = `
<input
  class="form__control"
  id="foo"
  type="hidden"
/>
`;
